Automattic/wp-calypso
# [Private sites] Redirect private atomic users to wp admin classic editor (from modal)

#### Changes proposed in this Pull Request

**Problem**

Classic editor in calypso does not support displaying private media files because of CORS issues. Adding a support would be super complex and probably not worth it, especially considering that we're phasing out the classic editor altogether. Classic editor in wp-admin handles private files out of the box so we're redirecting all private+atomic+classic editor users from calypso to wp-admin with `?classic_editor` attached to URL.

**Proposed solution**

We do have some dialog windows in place when user uses classic editor in Calypso:

Gutenberg content warning:

<img width="1090" alt="Zrzut ekranu 2020-03-13 o 16 32 06" src="https://user-images.githubusercontent.com/205419/76635769-3cc7d000-6548-11ea-8e48-ddbed3e6d744.png">

Gutenberg opt-in:

<img width="963" alt="Zrzut ekranu 2020-03-17 o 16 02 19" src="https://user-images.githubusercontent.com/205419/76869487-b291ab80-6868-11ea-965b-2156ecba5f6d.png">

My proposal is that for v1 we:
* Ensure that private+atomic user always sees one of these two modals when visiting classic editor. If no Gutenberg blocks are found in post content, let's display opt-in dialog.
* Ensure that choosing Classic editor redirects to corresponding editor in WP admin (ensured by 412-gh-wpcomsh).
* Make sure nothing changes for users who are not private+atomic+classic editor

#### Testing instructions

* Create a new private atomic site using a8c account and `calypso.localhost:3000` (start with simple site then go atomic - it should stay private)
* Apply related wpcomsh change from this PR **to the site you just created** 412-gh-wpcomsh
* Create a Gutenberg-based post, save it, in three dots menu select "Switch to classic editor"
* You should see Gutenberg content warning modal as on the screenshot above. Confirm that classic editor redirects to classic editor in wp-admin.
* Go back to calypso, create new page
* You should see Gutenberg opt-in modal as on the screenshot above. Confirm that choosing classic editor redirects to classic editor in wp-admin.
* Try on a non-private atomic site and confirm the behavior is the same as currently in production, e.g. you're not getting redirected to `wp-admin` and you're not getting opt-in modal every single time
